BlueprintUICommonControls/Sources/Rule.swift

```diff
@@ -0,0 +1,77 @@
+import BlueprintUI
+import UIKit
+
+/// A solid line, parallel to the x or y axis, with a fixed thickness but unconstrained in length,
+/// suitable for use as a separator.
+public struct Rule: ProxyElement {
+    /// Represents whether the rule is parallel to the x or y axis.
+    public enum Orientation {
+        /// Indicates that the rule is parallel to the x axis.
+        case horizontal
+        /// Indicates that the rule is parallel to the y axis.
+        case vertical
+    }
+
+    /// Represents the thickness of a rule in the direction perpendicular to its orientation.
+    public enum Thickness {
+        /// Indicates that the rule should be exactly 1 screen pixel thick,
+        /// the thinnest possible line that can be drawn.
+        case hairline
+        /// An exact thickness in points.
+        case points(CGFloat)
+
+        var value: CGFloat {
+            switch self {
+            case .hairline:
+                return 1.0 / UIScreen.main.scale
```

Comment:
My concern with magic behavior is mostly around this static reference to `UIScreen.main.scale`. Ideally we could set this explicitly but have a convenient default. See the way I implemented screen scale rounding in #18 for context.

One option is to add a screen scale property to `Rule`, but it doesn't make sense in cases where the thickness is a specific value.

Another option is to rewrite Thickness like this:
```swift
    public struct Thickness {
        public static func hairline(scale: CGFloat = UIScreen.main.scale) -> Thickness {
            return Thickness(value: 1.0 / scale)
        }

        public static func points(_ value: CGFloat) -> Thickness {
            return Thickness(value: value)
        }

        var value: CGFloat

        private init(value: CGFloat) {
            self.value = value
        }
    }
```

That makes it require parens at the point of use:

```swift
Rule(orientation: .horizontal, color: .black, thickness: .hairline())
```

But the autocomplete is pretty good and makes the default reasonably discoverable.

<img width="384" alt="Screen Shot 2019-10-10 at 4 27 46 PM" src="https://user-images.githubusercontent.com/100192/66613835-a7c94100-eb7b-11e9-8e87-84122abe6066.png">

Maybe that's fine?

Reply:
At the time when a developer is building a UI from elements they really should not have to worry about screen scale. If this information is required somewhere in the layout pass, it would be worth exploring how we expose this information (the current device/environment context) to the element implementation.

In the meantime, I'm ok using `UIScreen.main` as it will always be correct in practice.